tcf_website/models/models.py

```diff
@@ -690,23 +700,24 @@ def downvote(self, user):
     @staticmethod
     def display_reviews(course, instructor, user):
         """Prepare review list for course-instructor page."""
-
         reviews = Review.objects.filter(
             instructor=instructor,
             course=course,
-        ).exclude(text="").order_by("-created")
-
+        ).exclude(text="").annotate(
+            sum_votes=models.functions.Coalesce(
+                models.Sum('vote__value'),
+                models.Value(0)
+            ),
+        )
         if user.is_authenticated:
-            for review in reviews:
-                try:
-                    review.user_vote = Vote.objects.get(
-                        user=user,
-                        review=review,
-                    ).value
-                except Vote.DoesNotExist:
-                    review.user_vote = 0
-
-        return reviews
+            reviews = reviews.annotate(
+                user_vote=models.functions.Coalesce(
+                    models.Sum('vote__value',
+                               filter=models.Q(vote__user=user)),
+                    models.Value(0)
+                ),
```

Comment:
Just for my own curiosity: _why_ does this work? I assume it's just because it does it in one SQL query instead of a whole bunch?

Reply:
The original loop accesses the database for each review, whereas this query accesses the database only once. Also, this does not cause any exceptions--they incur a lot of overhead.